feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureToggles.kt

```diff
@@ -182,10 +191,27 @@ internal class ToggleImpl constructor(
     private val isInternalAlwaysEnabled: Boolean,
     private val appVersionProvider: () -> Int,
     private val flavorNameProvider: () -> String = { "" },
+    private val appVariantProvider: () -> String?,
 ) : Toggle {
+
+    private fun Toggle.State.isVariantTreated(variant: String?): Boolean {
+        // if no variants a present, we consider always treated
+        if (this.targets.isEmpty()) {
+            return true
+        }
+        // if variants present BUT no variant has been assigned yet, consider not treated
+        if (variant == null) {
+            return false
+        }
```

Comment:
I think I confused you when you asked me earlier. This should return `true`. 
If the variant hasn't been assigned when checking feature flag (experiment), we should consider it as treated and add the logic to assign the default one. Does it make sense?